### PR TITLE
Blueprint Development guide

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

This PR is to add a Blueprint development guide for new folks who would want to develop Blueprint for their application.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #1193 

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -563,3 +569,101 @@ and manually updating the action set to use artifacts created by the
...
+
+Blueprint Development
+=====================
+
+To start with Blueprint Development we need to be aware of some terms such as
+Actions, Phase, KanisterFunctions, Objects which has already been covered in
+this tutorial.
+
+Define and add ``actions`` for the blueprint as per your application. Each action
+performs a set of tasks. For example, backup, restore, and delete.
```
Comment (kale-amruta) [inline docs/tutorial.rst:581]: actions can be user defined right? Not necessarily backup/restore/delete. Can we specify that?

File: docs/tutorial.rst [hunk 2]
```diff
@@ -563,3 +569,101 @@ and manually updating the action set to use artifacts created by the
...
+An action can also consume information from Kubernetes objects
+such as :ref:`Secret<consumingSecret>` and :ref:`ConfigMap<consumingConfigMap>`
+
+To consume data from objects we need to reference it using ``configMapNames``
+and ``secretNames`` fields in an action.
+
+We use :ref:`templates` to access ConfigMap, Secrets, Artifacts data in
+the blueprint. See the example below.
+
+Here in this blueprint ``time-log-bp`` we have defined two actions
```
Comment (kale-amruta) [inline docs/tutorial.rst:605]: specify what is the functionality of this blueprint

File: docs/tutorial.rst [hunk 3]
```diff
@@ -563,3 +569,101 @@ and manually updating the action set to use artifacts created by the
...
+will hold a reference to that data.
+Artifacts created by phases in an action are Output Artifacts.
+Artifacts consumed by an action are Input Artifacts.
+
+An action can also consume information from Kubernetes objects
+such as :ref:`Secret<consumingSecret>` and :ref:`ConfigMap<consumingConfigMap>`
+
+To consume data from objects we need to reference it using ``configMapNames``
+and ``secretNames`` fields in an action.
+
```
Comment (kale-amruta) [inline docs/tutorial.rst:601]: We can also pass secrets as objects right?
```
  objects:
        pgSecret:
          kind: Secret
          name: '{{ .StatefulSet.Name }}'
          namespace: '{{ .StatefulSet.Namespace }}'
 ```

Comment (A-kanksh-a) [reply docs/tutorial.rst:601]: I think we pass these objects inside `phases` and not `action`.

```
type BlueprintAction struct {
	Name               string              `json:"name"`
	Kind               string              `json:"kind"`
	ConfigMapNames     []string            `json:"configMapNames,omitempty"`
	SecretNames        []string            `json:"secretNames,omitempty"`
	InputArtifactNames []string            `json:"inputArtifactNames,omitempty"`
	OutputArtifacts    map[string]Artifact `json:"outputArtifacts,omitempty"`
	Phases             []BlueprintPhase    `json:"phases,omitempty"`
}

```

```
type BlueprintPhase struct {
	Func       string                     `json:"func"`
	Name       string                     `json:"name"`
	ObjectRefs map[string]ObjectReference `json:"objects,omitempty"`
	Args       map[string]interface{}     `json:"args"`
}
```

**Pull request closed without merging**